src/main/java/samza/examples/azure/AzureBlobApplication.java

```diff
@@ -0,0 +1,136 @@
+ *     ./deploy/kafka/bin/kafka-console-producer.sh --topic page-view-azure-blob-input --broker-list localhost:9092 <br/>
+ *     {"userId": "user1", "country": "india", "pageId":"google.com"} <br/>
+ *     {"userId": "user2", "country": "france", "pageId":"facebook.com"} <br/>
+ *     {"userId": "user3", "country": "china", "pageId":"yahoo.com"} <br/>
+ *     {"userId": "user4", "country": "italy", "pageId":"linkedin.com"} <br/>
+ *     {"userId": "user5", "country": "germany", "pageId":"amazon.com"} <br/>
+ *     {"userId": "user6", "country": "denmark", "pageId":"apple.com"} <br/>
+ *   </li>
+ *   <li>
+ *    Seeing Output:
+ *    <ol>
+ *      <li>
+ *       See blobs in your Azure portal at https://<azure-storage-account-name>.blob.core.windows.net/oss-testcontainer/PageViewEventStream/<time-stamp>.avro
+ *      </li>
+ *      <li>
+ *       system-name "oss-testcontainer" in configs and code below maps to Azure-Container in Azure Storage account.
+ *      </li>
+ *      <li>
+ *       <time-stamp> is of the format yyyy/MM/dd/HH/mm-ss-randomString.avro. Hence navigate through the virtual folders on the portal to see your blobs.
+ *      </li>
+ *      <li>
+ *       Due to network calls, allow a few minutes for blobs to appear on the portal.
+ *      </li>
+ *      <li>
+ *       Config "maxMessagesPerBlob=2" ensures that a blob is created per 2 input messages. Adjust input or config accordingly.
+ *      </li>
+ *    </ol>
+ *   </li>
+ * </ol>
+ */
+public class AzureBlobApplication implements StreamApplication {
+  private static final List<String> KAFKA_CONSUMER_ZK_CONNECT = ImmutableList.of("localhost:2181");
+  private static final List<String> KAFKA_PRODUCER_BOOTSTRAP_SERVERS = ImmutableList.of("localhost:9092");
+  private static final Map<String, String> KAFKA_DEFAULT_STREAM_CONFIGS = ImmutableMap.of("replication.factor", "1");
+  private static final String INPUT_PAGEVIEW_STREAM_ID = "page-view-azure-blob-input";
+  private static final String OUTPUT_SYSTEM = "oss-testcontainer";
+  private static final String OUTPUT_STREAM = "PageViewEventStream";
+
+  @Override
+  public void describe(StreamApplicationDescriptor appDescriptor) {
+    // Define a system descriptor for Kafka
+    KafkaSystemDescriptor kafkaSystemDescriptor =
+        new KafkaSystemDescriptor("kafka").withConsumerZkConnect(KAFKA_CONSUMER_ZK_CONNECT)
+            .withProducerBootstrapServers(KAFKA_PRODUCER_BOOTSTRAP_SERVERS)
+            .withDefaultStreamConfigs(KAFKA_DEFAULT_STREAM_CONFIGS);
+
+    KafkaInputDescriptor<PageView> pageViewInputDescriptor =
+        kafkaSystemDescriptor.getInputDescriptor(INPUT_PAGEVIEW_STREAM_ID, new JsonSerdeV2<>(PageView.class));
+
+    // Define a system descriptor for Azure Blob Storage
+    GenericSystemDescriptor azureBlobSystemDescriptor =
+        new GenericSystemDescriptor(OUTPUT_SYSTEM, "org.apache.samza.system.azureblob.AzureBlobSystemFactory");
```

Comment:
It might be nice to have a system descriptor for the azure blob system, but probably not necessary now. The benefit is that you could move some of the properties into this file directly.